pkg/fqdn/name_manager_test.go

```diff
@@ -147,3 +148,46 @@ func TestNameManagerMultiIPUpdate(t *testing.T) {
 	nameManager.Unlock()
 
 }
+
+func TestActiveConnectionsTTLUpdate(t *testing.T) {
+	now := time.Now()
+	dnsZombies := NewDNSZombieMappings(defaults.ToFQDNsMaxDeferredConnectionDeletes, defaults.ToFQDNsMaxIPsPerHost)
+	sharedIP := netip.MustParseAddr("1.1.1.1")
+	dnsZombies.Upsert(now, sharedIP, "cilium.io")
+
+	var (
+		activeConnectionTTL = 2
+		nameManager         = NewNameManager(Config{
+			MinTTL:               1,
+			Cache:                NewDNSCache(0),
+			ActiveConnectionsTTL: activeConnectionTTL,
+			IPCache:              testipcache.NewMockIPCache(),
+
+			UpdateSelectors: func(ctx context.Context, selectorIPMapping map[api.FQDNSelector][]netip.Addr, _ uint64) *sync.WaitGroup {
+				return &sync.WaitGroup{}
+			},
+			GetEndpointsDNSInfo: func(_ string) []EndpointDNSInfo {
+				return []EndpointDNSInfo{
+					{
+						ID:         "1",
+						DNSHistory: NewDNSCache(0),
+						DNSZombies: dnsZombies,
+					},
+				}
+			}})
+	)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// GC should return 1 zombie with updated TTL
+	nameManager.GC(ctx)
+	for _, v := range nameManager.cache.forward {
+		for _, vv := range v {
+			expectedHr, expectedMin, expectedSec := time.Now().Add(time.Duration(activeConnectionTTL) * time.Second).Clock()
+			actualHr, actualMin, actualSec := vv.ExpirationTime.Clock()
+			require.Equal(t, expectedHr, actualHr)
+			require.Equal(t, expectedMin, actualMin)
+			require.Equal(t, expectedSec, actualSec)
+		}
+	}
```

Comment:
This test is not doing anything. The `nameManager.cache.forward` slice is empty.

Reply:
Why is that ? I am seeing the length of cache.forward being 1 after adding a break point.
![image](https://github.com/cilium/cilium/assets/8600441/d4e55902-1ce5-41b6-99f9-fe97a12b1c7d)

Reply:
I pulled the PR fresh and it looks like I was mistaken, sorry for the noise. The NameManager cache starts empty, but since the endpoint has a pointer to a list of zombies where the zombie's "active" time is set to the start of the test, the GC call then pulls those zombies into the NameManager with the newly introduced active connection timeout.